packages/datetime/src/datePicker.tsx

```diff
@@ -198,7 +230,15 @@ export class DatePicker extends AbstractComponent<IDatePickerProps, IDatePickerS
         );
     }
 
-    private handleDayClick = (day: Date, modifiers: ReactDayPicker.DayModifiers) => {
+    private handleDayClick = (
+        day: Date,
+        modifiers: ReactDayPicker.DayModifiers,
+        e: React.MouseEvent<HTMLDivElement>,
+    ) => {
+        if (typeof this.props.dayPickerProps.onDayClick === "function") {
```

Comment:
We have a helper function for this:

```tsx
import { Utils as CoreUtils } from "@blueprintjs/core";

// ...

CoreUtils.safeInvoke(this.props.dayPickerProps.onDayClick, day, modifiers, e);
```

We also tend to invoke custom callbacks at the _end_ of the function, but I'm worried that the `setState`s and `forceUpdate`s might complicate that approach. @giladgray thoughts?